mdstcpip/mdsipshr/ConnectToMds.c

```diff
@@ -82,7 +82,7 @@ static int do_login(Connection *c)
   int status = SendMdsMsgC(c, msend, 0);
   int err;
   free(msend);
-  if (STATUS_NOT_OK)
+  if ((status == SsINTERNAL) || STATUS_NOT_OK)
```

Comment:
In the other changes, we rewrite `status` to `MDSplusERROR`, but in this one we just check for `SsINTERNAL` as well, is there a reason for the difference?

Reply:
No, I don't have a good reason.   Other than the changes were made over several days, thus I was inconsistent.

The longer answer . . . 

I'm taking a cautious approach until we all reach agreement on the broader SsINTERNAL problem, Issue #2741. 

Timo tells me it is important to keep SsINTERNAL so that it ripples up the call stack.   However, most of the code I've looked at never checks for SsINTERNAL.   This is a systemic error throughout all of MDSplus.   This PR is just a targeted fix for the issues I saw while investigating the dispatcher's leaking sockets.

My preference would be to redefine SsINTERNAL so it is a valid MDSplus error code (i.e., set it to `-2` so that the low order bit is zero, meaning failure).   Timo and I have not reached agreement on whether that is feasible or not.